src/Persistence/Array_.php

```diff
@@ -328,72 +328,65 @@ protected function setLimitOrder($m, &$action)
      *
      * @return \atk4\data\Action\Iterator|null
      */
-    public function applyConditions(Model $m, \atk4\data\Action\Iterator $q)
+    public function applyConditions(Model $model, \atk4\data\Action\Iterator $iterator)
     {
-        if (!isset($m->conditions)) {
+        if (empty($model->conditions)) {
             // no conditions are set in the model
-            return $q;
+            return $iterator;
         }
 
-        foreach ($m->conditions as $cond) {
+        foreach ($model->conditions as $cond) {
 
-            // Options here are:
-            // count($cond) == 1, we will pass the only
-            // parameter inside where()
+            // assume the action is "where" if we have only 2 parameters
+            if (count($cond) == 2) {
+                array_splice($cond, -1, 1, ['where', $cond[1]]);
```

Comment:
why not just `$cond = [$cond[0], 'where', $cond[1]]` ? :) but of course this works too.